app/views/tasks/show.html.slim

```diff
@@ -561,17 +560,39 @@
         = link_to task_path(@task), class: 'btn btn-important', method: :delete, data: {confirm: t('common.sure')} do
           i.fa-solid.fa-trash-can
           =< t('common.button.delete')
-
-      - if policy(@task).duplicate?
-        = link_to duplicate_task_path(@task), method: :post, class: 'btn btn-important' do
-          i.fa-regular.fa-clone
-          =< t('common.button.duplicate')
+      - if @task.contribution?
+        - if policy(@task.parent).edit?
+          = link_to approve_changes_task_task_contribution_path(@task.parent, @task.task_contribution), method: :post, class: 'btn btn-important' do
+            i.fa-solid.fa-check
+            =< t('common.button.approve')
+          = link_to discard_changes_task_task_contribution_path(@task.parent, @task.task_contribution), method: :post, class: 'btn btn-important' do
+            i.fa-solid.fa-xmark
+            =< t('common.button.reject')
+        - else
+          = link_to discard_changes_task_task_contribution_path(@task.parent, @task.task_contribution), method: :post, class: 'btn btn-important' do
+            i.fa-solid.fa-link-slash
+            =< t('task_contributions.buttons.decouple')
       - else
-        div data-bs-toggle='tooltip' title=t('.guest.disabled.tooltip') data-bs-delay=150
-          = link_to duplicate_task_path(@task), method: :post, class: 'btn btn-important disabled' do
+        - if policy(@task).contribute?(check_other_contrib: false)
+          - if @task.contributions(user: @current_user).any?
+            = link_to edit_task_task_contribution_path(@task, @task.contributions(user: @current_user).first), method: :get, class: 'btn btn-important' do
+              i.fa-regular.fa-edit
+              =< t('task_contributions.buttons.edit_contribution')
+          - else
+            = link_to new_task_task_contribution_path(@task), method: :get, class: 'btn btn-important' do
+              i.fa-regular.fa-edit
+              =< t('task_contributions.buttons.contribute')
+
+        - if policy(@task).duplicate?
+          = link_to duplicate_task_path(@task), method: :post, class: 'btn btn-important' do
+
             i.fa-regular.fa-clone
             =< t('common.button.duplicate')
-
+        - else
+          div[data-bs-toggle='tooltip' title=t('.guest.disabled.tooltip') data-bs-delay=150]
+            = link_to duplicate_task_path(@task), method: :post, class: 'btn btn-important disabled' do
+              i.fa-regular.fa-clone
+              =< t('common.button.duplicate')
```

Comment:
Can we add a disabled button for the task contribution as well? I think, it is free marketing for the feature 🙂